openc3/lib/openc3/api/settings_api.rb

```diff
@@ -35,41 +35,36 @@ module Api
 
     def list_settings(scope: $openc3_scope, token: $openc3_token)
       authorize(permission: 'system', scope: scope, token: token)
-      SettingsModel.names(scope: scope)
+      SettingModel.names(scope: scope)
     end
 
     def get_all_settings(scope: $openc3_scope, token: $openc3_token)
       authorize(permission: 'system', scope: scope, token: token)
-      SettingsModel.all(scope: scope)
+      SettingModel.all(scope: scope).transform_values { |hash| JSON.parse(hash['data'], :allow_nan => true, :create_additions => true) }
     end
 
     def get_setting(name, scope: $openc3_scope, token: $openc3_token)
       authorize(permission: 'system', scope: scope, token: token)
-      setting = SettingsModel.get(name: name, scope: scope)
-      if setting
-        return setting["data"]
+      result = SettingModel.get(name: name, scope: scope)
+      if result
+        JSON.parse(result['data'], :allow_nan => true, :create_additions => true)
       else
-        return nil
+        nil
       end
     end
 
-    def get_settings(*args, scope: $openc3_scope, token: $openc3_token)
+    def get_settings(*settings, scope: $openc3_scope, token: $openc3_token)
       authorize(permission: 'system', scope: scope, token: token)
-      ret = []
-      args.each do |name|
-        setting = SettingsModel.get(name: name, scope: scope)
-        if setting
-          ret << setting["data"]
-        else
-          ret << nil
-        end
-      end
-      return ret
+      result = []
+      settings.each { |name| result << get_setting(name, scope: scope, token: token) }
+      result
     end
 
     def save_setting(name, data, scope: $openc3_scope, token: $openc3_token)
       authorize(permission: 'admin', scope: scope, token: token)
-      SettingsModel.set({ name: name,  data: data }, scope: scope)
+      json = JSON.generate(data.as_json(:allow_nan => true))
```

Comment:
I decided to JSON.generate and then JSON.parse on all the data going in and out. Solved some problems and it matches what we're doing in stash_api.rb.

Reply:
Will that break existing stuff?

Reply:
Probably. If you had a bunch of banner settings for example I think they were previously just being stored as a big string. Now it's generated JSON. Seems like before the end user was deciding whether to store JSON or simple strings or whatever. That makes it hard to work with in local_mode because there isn't consistency. I guess I can just save it off as is in local_mode as well.

Reply:
You can create a migration if it does break things.